### PR TITLE
[go][tools] Add Kotlin support to ReactAndroidCodeTransformer

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

With the Kotlinfy effort on react-native, some of the files used by our `ReactAndroidCodeTransformer ` were converted to Kotlin

Related to ENG-12562

# How

Update `ReactAndroidCodeTransformer` to support Kotlin 

# Test Plan

run `et update-rn`

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: apps/expo-go/android/tools/build.gradle [hunk 1]
```diff
@@ -2,6 +2,9 @@ apply plugin: 'java'
 
 dependencies {
   implementation fileTree(dir: 'libs', include: ['*.jar'])
+  implementation "org.jetbrains.kotlin:kotlin-compiler-embeddable:1.8.0"
+  implementation "org.jetbrains.kotlin:kotlin-compiler:1.8.0"
+  implementation "org.jetbrains.kotlin:kotlin-scripting-compiler-embeddable:1.8.0"
```
Comment (Kudo) [inline apps/expo-go/android/tools/build.gradle:7]: quick question: do we need these for ReactAndroidCodeTransformer? wondering if that would bring to runtime cost?

File: apps/expo-go/android/tools/src/main/java/host/exp/exponent/tools/ReactAndroidCodeTransformer.java [hunk 2]
```diff
@@ -251,72 +259,41 @@ public Node visit(String methodName, MethodDeclaration n) {
...
-            return n;
-          case "isHotModuleReplacementEnabled":
-            n.setPublic(true);
-            return n;
-          case "setHotModuleReplacementEnabled":
-            n.setPublic(true);
-            return n;
-          case "getPackagerConnectionSettings":
-            n.setPublic(true);
-            return n;
```
Comment (gabrieldonadel) [inline apps/expo-go/android/tools/src/main/java/host/exp/exponent/tools/ReactAndroidCodeTransformer.java:281]: WIth the Kotlinfy effort we no longer need to set these as public https://github.com/facebook/react-native/pull/43731

File: apps/expo-go/android/tools/build.gradle (hunk 1 above)
Comment (gabrieldonadel) [reply apps/expo-go/android/tools/build.gradle]: Actually, this won't be necessary for now, taking a deeper look at the Kotlin files I notice that we won't need the visitor function for Kotlin files for now, we can keep with just using the modifySource

**Review by Kudo — APPROVED** [review]
nice explanation for each changes. thanks for that.

**Pull request merged**